docs/gadgets/run.md

```diff
@@ -57,6 +57,45 @@ ubuntu-hirsute         default                mypod2                 mypod2
 ubuntu-hirsute         default                mypod2                 mypod2                 242164  cat         0        0        3   /dev/null
 ```
 
+### Private registries in Kubernetes
+
+In order to use private registries, you will need a [Kubernetes secret](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/) with name `gadget-pull-secret` having credentials to access the registry.
+The secret should be in the same namespace as `gadget` pod so start by creating namespace `gadget`, if it doesn't exist:
+
+```bash
+$ kubectl create namespace gadget
+```
+
+then create the secret:
+
+```bash
+$ kubectl create secret docker-registry gadget-pull-secret -n gadget --docker-server=MYSERVER --docker-username=MYUSERNAME --docker-password=MYPASSWORD
+````
+
+or you can create the secret from a file:
+
+```bash
+$ kubectl create secret docker-registry regcred -n gadget --from-file=.dockerconfigjson=$HOME/.docker/config.json
+```
+
+the secret can be used in following two ways while:
+
+#### running a gadget
+
+```bash
+$ kubectl gadget run ghcr.io/inspektor-gadget/gadget/trace_tcpconnect:latest --pull-secret
```

Comment:
nit: it'll be better to use something like `myprivateregistry.io/trace_tcpconnect:latest` (just to be sure people understand it's about gadgets on private registries, but it's not needed for the gadgets we expose as they are public)